modules/globebrowsing/src/geojson/geojsoncomponent.cpp

```diff
@@ -0,0 +1,743 @@
+        CentroidCoordinateInfo,
+        glm::vec2(0.f),
+        glm::vec2(-90.f, -180.f),
+        glm::vec2(90.f, 180.f)
+    )
+    , _flyToFeature(FlyToFeatureInfo)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _enabled = p.enabled.value_or(_enabled);
+    addProperty(_enabled);
+
+    _opacity = p.opacity.value_or(_opacity);
+    addProperty(_opacity);
+    addProperty(_fade);
+
+    _geoJsonFile = p.file.string();
+    _geoJsonFile.setReadOnly(true);
+    addProperty(_geoJsonFile);
+
+    _ignoreHeightsFromFile = p.ignoreHeights.value_or(_ignoreHeightsFromFile);
+
+    const float minGlobeRadius = static_cast<float>(
+        _globeNode.ellipsoid().minimumRadius()
+    );
+
+    _heightOffset = p.heightOffset.value_or(_heightOffset);
+    _heightOffset.onChange([this]() { _heightOffsetIsDirty = true; });
+    _heightOffset.setMinValue(-0.9f * minGlobeRadius);
+    _heightOffset.setMaxValue(5.f * minGlobeRadius);
+    addProperty(_heightOffset);
+
+    _latLongOffset = p.coordinateOffset.value_or(_latLongOffset);
+    _latLongOffset.onChange([this]() { _dataIsDirty = true; });
+    addProperty(_latLongOffset);
+
+    if (p.defaultProperties.has_value()) {
+        _defaultProperties.createFromDictionary(*p.defaultProperties);
+    }
+    addPropertySubOwner(_defaultProperties);
+
+    _defaultProperties.pointTexture.onChange([&]() {
+        std::filesystem::path texturePath = _defaultProperties.pointTexture.value();
+        // Not ethat an empty texture is also valid => use default texture from module
+        if (std::filesystem::is_regular_file(texturePath) || texturePath.empty()) {
+            _textureIsDirty = true;
+        }
+        else {
+            LERROR(fmt::format(
+                "Provided texture file does not exist: '{}'",
+                _defaultProperties.pointTexture
+            ));
+        }
+    });
+
+    _defaultProperties.shouldTesselate.onChange([&]() { _dataIsDirty = true; });
+    _defaultProperties.tesselationLevel.onChange([&]() { _dataIsDirty = true; });
+    _defaultProperties.tesselationMaxDistance.onChange([&]() { _dataIsDirty = true; });
+
+    _forceUpdateHeightData.onChange([this]() {
+        for (GlobeGeometryFeature& f : _geometryFeatures) {
+            f.updateHeightsFromHeightMap();
+        }
+    });
+    addProperty(_forceUpdateHeightData);
+
+    addProperty(_preventUpdatesFromHeightMap);
+
+    _drawWireframe = p.drawWireframe.value_or(_drawWireframe);
+    addProperty(_drawWireframe);
+
+    using PointRenderMode = GlobeGeometryFeature::PointRenderMode;
+    _pointRenderModeOption.addOptions({
+        { static_cast<int>(PointRenderMode::AlignToCameraDir), "AlignToCameraDir"},
+        { static_cast<int>(PointRenderMode::AlignToCameraPos), "AlignToCameraPos"},
+        { static_cast<int>(PointRenderMode::AlignToGlobeNormal), "AlignToGlobeNormal"},
+        { static_cast<int>(PointRenderMode::AlignToGlobeSurface), "AlignToGlobeSurface"}
```

Comment:
Yes, of course 👍 They're not going ot show up clearly in the GUI though, because of their length... 

can we somehow rename the property to get rid of the "Align to" part, I wonder 🤔

Reply:
Good point

How about renaming the property gui name from "Point Render Mode" to "Points Aligned to" and then call the options: "Camera Direction", "Camera Position", "Globe Normal", and "Globe Surface"

Reply:
I like that idea, but then I should probably change so that codegen expects those as keys for the option mapping as well

Reply:
Did that